examples/msgpack/msgpack_entity.go

```diff
@@ -0,0 +1,42 @@
+package restPack
+
+import (
+	restful "github.com/emicklei/go-restful"
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+const MIME_MSGPACK = "application/x-msgpack" // Accept or Content-Type used in Consumes() and/or Produces()
+
+// NewEntityAccessorMPack returns a new EntityReaderWriter for accessing MessagePack content.
+// This package is not initialized with such an accessor using the MIME_MSGPACK contentType.
+func NewEntityAccessorMsgPack(contentType string) restful.EntityReaderWriter {
+	return entityMsgPackAccess{ContentType: contentType}
+}
+
+// entityOctetAccess is a EntityReaderWriter for Octet encoding
+type entityMsgPackAccess struct {
+	// This is used for setting the Content-Type header when writing
+	ContentType string
```

Comment:
this is needed by unit test that can check the content-type.

i found a problem.
in my case, i have called response.writeAsJson to response client message,
but if i want response data with message pack format, i cannot do this, because response.go not contain any method about that...

i think i must folk your project then add this method then reference this folk that i can response client with message pack format.
```
func (r *Response) WriteAsMsgPack(value interface{}) error {
	return writeAsMsgPack(r, http.StatusOK, MIME_MSGPACK, value)
}
```
do you have any comments?